src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -2528,8 +2532,33 @@ bool RTPSParticipantImpl::is_reader_ignored(
 }
 
 bool RTPSParticipantImpl::ignore_participant(
-        const GuidPrefix_t& /*participant_guid*/)
+        const GuidPrefix_t& participant_guid)
 {
+    {
+        shared_lock<eprosima::shared_mutex> _(mp_builtinProtocols->getDiscoveryMutex());
+
+        for (auto server_it = m_att.builtin.discovery_config.m_DiscoveryServers.begin();
+                server_it != m_att.builtin.discovery_config.m_DiscoveryServers.end(); server_it++)
+        {
+            if (server_it->guidPrefix == participant_guid)
+            {
+                EPROSIMA_LOG_WARNING(RTPS_PARTICIPANT, "Cannot ignore one of this participant Discovery Servers");
+                return false;
+            }
+        }
+    }
+    if (!is_participant_ignored(participant_guid))
+    {
+        {
+            {
+                std::unique_lock<shared_mutex> _(ignored_mtx_);
+                ignored_participants_.insert(participant_guid);
+            }
+            pdp()->remove_remote_participant(GUID_t(participant_guid, c_EntityId_RTPSParticipant),
+                    ParticipantDiscoveryInfo::DISCOVERY_STATUS::IGNORED_PARTICIPANT);
+        }
+        return true;
+    }
     return false;
```

Comment:
Ignoring a participant twice should not be an error, so I think we can always return true here.

```suggestion
    }
    return true;
```